File: frontend/src/api/ai.ts
import api from './client'

export type AISummaryResponse = {
  summary: string
  highlights: string[]
  warnings: string[]
  recommendations: string[]
  metadata: {
    month: number
    year: number
    income: number
    expense: number
    balance: number
    savings_rate: number
    top_category: string | null
    top_category_amount: number
    expense_change_percent: number
  }
}

export type AIChatResponse = {
  answer: string
  intent: string
  metadata: Record<string, unknown>
}

export async function generateAISummary(month: number, year: number) {
  const response = await api.post<AISummaryResponse>('/ai/summary', {
    month,
    year,
  })

  return response.data
}

export async function sendChatMessage(message: string) {
  const response = await api.post<AIChatResponse>('/ai/chat', {
    message,
  })

  return response.data
}

File: backend/app/schemas/ai.py
from pydantic import BaseModel, Field


class AISummaryRequest(BaseModel):
    month: int = Field(..., ge=1, le=12)
    year: int = Field(..., ge=2000, le=2100)


class AISummaryResponse(BaseModel):
    summary: str
    highlights: list[str]
    warnings: list[str]
    recommendations: list[str]
    metadata: dict


class AIChatRequest(BaseModel):
    message: str = Field(..., min_length=2, max_length=500)


class AIChatResponse(BaseModel):
    answer: str
    intent: str
    metadata: dict